src/bitski.ts

```diff
@@ -144,14 +144,14 @@ export class Bitski {
     return OAuthProviderIntegrationType.REDIRECT;
   }
 
-  private createEngine(fetchSubprovider: Subprovider, authorizationSubprovider: AuthorizationHandler, options?: any): ProviderEngine {
+  private createEngine(subproviders: Subprovider[], options?: any): ProviderEngine {
     const engine = new ProviderEngine(options);
 
     this.addDefaultSubproviders(engine);
 
-    engine.addProvider(authorizationSubprovider);
-
-    engine.addProvider(fetchSubprovider);
+    subproviders.forEach(subprovider => {
```

Comment:
could probably be simplified to `subproviders.forEach(engine.addProvider)`

Reply:
Good call. tweaked it.